src/compressor.py

```diff
@@ -107,16 +113,27 @@ def run_command(self, command, result_item):
                 if new_file_data.is_file():
                     result_item.new_size = new_file_data.stat().st_size
 
-                    # This check is mainly for compressors that don't have a way
-                    # to automatically detect and skip files
-                    if result_item.new_size > result_item.size:
-                        if self.do_new_file:
+                    # Manually skip files if necessary
+                    if get_file_type(result_item.original_filename) in ["webp", "svg"]:
+                        if self.do_new_file and result_item.new_size > result_item.size:
                             shutil.copy2(result_item.filename, result_item.new_filename)
-                        else:
-                            shutil.copy2(temp_filename, result_item.new_filename)
-                            Path(temp_filename).unlink(True)
-                        result_item.new_size = new_file_data.stat().st_size
-
+                            result_item.new_size = new_file_data.stat().st_size
+                            result_item.skipped = True
+                        elif not self.do_new_file:
+                            if not result_item.new_size > result_item.size:
+                                shutil.copy2(result_item.filename, result_item.original_filename)
+                            else:
+                                result_item.skipped = True
+
+                            # Remove temporary file
+                            Path(result_item.filename).unlink(True)
+                            result_item.filename = result_item.original_filename
+                            result_item.new_filename = result_item.original_filename
+                            new_file_data = Path(result_item.new_filename)
+                            result_item.new_size = new_file_data.stat().st_size
+                    else:
+                        if filecmp.cmp(result_item.filename, result_item.new_filename):
```

Comment:
What does the filecmp.cmp call check please here?
It's to see if something has been done on the file? Better than checking size right?

Reply:
Maybe else/if can be combined in an elif?

Reply:
The filecmp.cmp check compares the data returned by the os.stat function, which checks a bunch of different file properties, including file size. However, in hindsight I don't think it's very necessary to have to check all of that. I think I was just being overly cautious, and it's likely not worth importing an entire module when the file size can just be compared. 

Good catch on the else/if. I'll fix that in another commit.